### ISSUE TITLE
Add support for execing multiple python statements

### ISSUE THREAD

**Issue by bsteffensmeier** [ninia/jep#55]
Jep has the ability to eval a single python statement and it has the ability to run a script from a file but if a script is loaded into a Java String then there is no way to reliably execute the script. Jep should provide an exec method that matches the behavior of the python builtin exec and allows the caller to exec an entire script from a String.

Here are the workarounds I have experimented with to get around this missing capability:
1. The most reliable way is to simply exec the entire script in python through something like this:

``` java
jep.set("_pythonScript", script.toString());
jep.eval("exec _pythonScript");
jep.eval("del _pythonScript");
```

or

``` java
jep.eval("exec \"\"\"" + pythonScript + "\"\"\"");
```
1. If the python script only contains a single python statement, such as a function or a class definition then it is possible to eval the entire script. This actually causes confusion because things seem to work for simple test cases but eventually a more complicated script breaks it. Even if there aren't multiple python statements it is still possible for things like trailing whitespace to cause syntax errors
2. If jep is in interactive mode you can feed the script to jep one line at a time. This breaks if there are blank lines in indented blocks because the block is evaluated early. Also if there are no blank lines between unindented lines then only the first statement is evaluated.

**Comment by ndjensen** [comment]
I'm not convinced `eval(String)` is written as well as it could be, in particular [this area](https://github.com/mrj0/jep/blob/master/src/jep/Jep.java#L465).  We might be able to fix it there.

Also some of the code in the C could possibly be improved [here](https://github.com/mrj0/jep/blob/master/src/jep/pyembed.c#L818).  In particular I'm not sure `Py_single_input` is the most appropriate option there.  One of [these other options](https://docs.python.org/2/c-api/veryhigh.html#c.Py_eval_input) might be better.

**Comment by svensk-pojke** [comment]
Hello. I am using Jep to execute python scripts from within a Java application. I have tried both methods that bsteffensmeier outlines above. Even with the numpy work around using shared modules I still get funny behavior. See the test case below.

```
import jep.Jep;
import jep.JepConfig;
public class JepTest {

	private static String script
		= "import numpy as np\n"
		+ "from skimage.io import imread\n"
		+ "testImagePath = '<path to test image>'\n"
		+ "testImage = imread(testImagePath)\n";

	private static JepConfig jepConfig = null;
	private static final String[] sharedModules = {"numpy", "scipy", "skimage", "skimage.transform", "skimage.io"};
	
	public static void main(String[] args) {

		try {
			jepConfig = new JepConfig();
			jepConfig.addSharedModules(sharedModules);
			Jep jep = new Jep(jepConfig);
			//jep.set("_pythonScript",  script);
			//jep.eval("exec _pythonScript");
			//jep.eval("del _pythonScript");
			jep.eval("exec \"\"\"" + script + "\"\"\"");
			System.out.println("Finished with script...");						
			jep.close();
		}
		catch (Exception e) {
			System.out.println("Caught Jep Exception:");
			e.printStackTrace();
		}
	}
}
```
This results in the following error:
```
Caught Jep Exception:
jep.JepException: <type 'exceptions.IOError'>: file() constructor not accessible in restricted mode
	at /usr/lib/python2.7/dist-packages/PIL/Image.open(Image.py:1996)
	at /usr/lib/python2.7/dist-packages/skimage/io/_plugins/pil_plugin.imread(pil_plugin.py:21)
	at /usr/lib/python2.7/dist-packages/skimage/io/_plugins/plugin.call(plugin.py:105)
	at /usr/lib/python2.7/dist-packages/skimage/io/_io.imread(_io.py:150)
```
Evidently the jep sub-interpreter is running in restricted execution mode. Any ideas on how to circumvent this problem are appreciated. Thanks.

**Comment by svensk-pojke** [comment]
Sorry, this is/was my first post. Thanks, I will look at #62.

**Comment by ndjensen** [comment]
Restricted mode issue split off to #68.

**Comment by tcederquist** [comment]
@bsteffensmeier, this isn't a complete fix but I have found it more reliable for interpreted engines to set the code as a variable and then execute it such as with a String (space indents and /r in the variable 'code'):

        python.set("__my_priv_var_code", code);
        python.eval("exec(__my_priv_var_code)");
        python.eval("del __my_priv_var_code");

**Comment by bsteffensmeier** [comment]
THis has been fixed with 725f5fe in the dev_3.9 branch and can be closed when Jep 3.9 is released.

**Comment by bsteffensmeier** [comment]
Jep 3.9.0 has been released.
